packages/data-point/lib/entity-types/base-entity/factory.js

```diff
@@ -5,30 +5,76 @@ const { normalizeTypeCheckSource } = require('../../helpers/type-check-helpers')
 const createReducer = require('../../reducer-types').create
 
 /**
- * @param {Function} Factory - factory function to create the entity
- * @param {Object} spec - spec for the Entity
- * @param {string} id - Entity's id
+ * @throws When function is not valid
+ * @param {Function} resolve
+ * @returns {Boolean} true if valid
  */
-function create (Factory, spec, id) {
-  const entity = new Factory(spec)
-
-  entity.id = id
-
-  if (spec.before) {
-    entity.before = createReducer(spec.before)
+function validateResolve (resolve) {
+  if (typeof resolve !== 'function') {
+    throw new Error(
+      `"resolve" must be a function, create(type:String, factory:Function, resolve:Function)`
+    )
   }
 
-  if (spec.value) {
-    entity.value = createReducer(spec.value)
+  if (resolve.length !== 2) {
+    throw new Error(
+      '"resolve" function must have an arity of 2, resolve(accumulator:Accumulator, resolveReducer:Function)'
+    )
   }
 
-  if (spec.after) {
-    entity.after = createReducer(spec.after)
+  return true
+}
+
+/**
+ * @throws When function is not valid
+ * @param {Function} factory
+ * @returns {Boolean} true if valid
+ */
+function validateFactory (factory) {
+  if (typeof factory !== 'function') {
+    throw new Error(
+      '"factory" argument must be a function, create(type:String, factory:Function, resolve:Function)'
+    )
   }
+  if (factory.length !== 2) {
+    throw new Error(
+      '"factory" argument must be a function with arity of 2 factory(name:String, spec:Object)'
```

Comment:
I prefer the wording of the previous (similar) error message :)

`'"resolve" function must have an arity of 2, resolve(accumulator:Accumulator, resolveReducer:Function)'`